src/frontend-samples/explode-sample/ExplodeTile.ts

```diff
@@ -0,0 +1,629 @@
+  transformWorld: Transform;
+}
+
+/** Information returned by the query for the elements to the backend. */
+interface ElementDataProps {
+  id: string;
+  origin: XYZProps;
+  bBoxHigh: XYZProps;
+  bBoxLow: XYZProps;
+  pitch: number;
+  roll: number;
+  yaw: number;
+}
+
+/** Parameters used to create a ElementTile. */
+interface TileParams {
+  centerOfMass: Point3d;
+  versionInfo: TileVersionInfo;
+  data: ElementData;
+}
+
+/** Identifies our TileTree for the supplier. */
+interface ExplodeTreeId {
+  name: string;
+  ids: string[];
+}
+
+/** Parameters for creating a Tile Tree */
+interface ExplodeTreeParams {
+  objectName: string;
+  data: ElementData[];
+  iModel: IModelConnection;
+  tileVersionInfo: TileVersionInfo;
+}
+
+/** Parameters for creating the overloaded draw argument. */
+interface ExplodeTileDrawArgsParams extends TileDrawArgParams {
+  explodeFactor: number;
+}
+
+/** Overloaded TileDrawArgs that also contain the current Explode Scaling Factor. */
+class ExplodeTileDrawArgs extends TileDrawArgs {
+  public explodeFactor: number;
+  constructor(params: ExplodeTileDrawArgsParams) {
+    super(params);
+    this.explodeFactor = params.explodeFactor;
+  }
+}
+
+/** Creates a transform matrix translating the range away from the point scaled. */
+function calculateExplodeTransform(centerOfElement: Point3d, displaceOrigin: Point3d, explodeScaling: number) {
+  const vector = Vector3d.createFrom(displaceOrigin);
+  vector.subtractInPlace(centerOfElement);
+  vector.scaleInPlace(explodeScaling - 1);
+  return Transform.createTranslation(vector).inverse()!;
+}
+
+/** Creates a Range3d containing all the points using the Range3d API. */
+function getRangeUnion(ranges: Range3d[]): Range3d {
+  const range = Range3d.createNull();
+  for (const r of ranges)
+    range.extendRange(r);
+  return range;
+}
+
+/** Handles the creation and getting of our TileTree for the TileTreeReference. */
+class ExplodeTreeSupplier implements TileTreeSupplier {
+  public ids: string[] = [];
+  private _info: TileVersionInfo | undefined;
+
+  /** Compare two tree Ids returning a negative number if lhs < rhs, a positive number if lhs > rhs, or 0 if the Ids are equivalent. */
+  public compareTileTreeIds(lhs: ExplodeTreeId, rhs: ExplodeTreeId): number {
+    let rtn = rhs.ids.length - lhs.ids.length;
+    // First compare lengths for speed, then compare object names.
+    if (rtn === 0)
+      rtn = compareStrings(rhs.name, lhs.name);
+    return rtn;
+  }
+
+  /** Produce the TileTree corresponding to the specified tree Id. The returned TileTree will be associated with its Id in a Map. */
+  public async createTileTree(id: ExplodeTreeId, iModel: IModelConnection): Promise<ExplodeTileTree | undefined> {
+    if (id.ids.length <= 0)
+      return undefined;
+
+    let info: Promise<TileVersionInfo>;
+    if (this._info === undefined)
+      info = IModelTileRpcInterface.getClient().queryVersionInfo();
+    else
+      info = Promise.resolve(this._info);
```

Comment:
This is awkward. Why not simply:
```
if (!this._info)
  this._info = await IModelTileRpcInterface.getClient().queryVersionInfo();
```
or query the version info and pass it into the supplier's constructor?

Reply:
It was organized this way to take advantage of async functions.  As the effects are unnoticeable, I have simplified as suggested.